tests/test-libarchive-import.c

```diff
@@ -243,6 +301,166 @@ test_libarchive_ignore_device_file (gconstpointer data)
   g_assert_no_error (error);
 }
 
+static void
+test_libarchive_ostree_convention (gconstpointer data)
+{
+  TestData *td = (void*)data;
+  GError *error = NULL;
+  struct archive *a = archive_read_new ();
+  OstreeRepoImportArchiveOptions opts = { 0, };
+
+  if (skip_if_no_xattr (td))
+    goto out;
+
+  test_archive_setup (td->fd, a);
+
+  opts.autocreate_parents = TRUE;
+  opts.use_ostree_convention = TRUE;
+  opts.ignore_unsupported_content = TRUE;
+
+  if (!import_write_and_ref (td->repo, &opts, a, "bar", NULL, &error))
+    goto out;
+
+  /* check contents */
+  if (!spawn_cmdline ("ostree --repo=repo ls bar file", &error))
+    goto out;
+
+  if (!spawn_cmdline ("ostree --repo=repo ls bar anotherfile", &error))
+    goto out;
+
+  if (!spawn_cmdline ("ostree --repo=repo ls bar /usr/etc/file", &error))
+    goto out;
+
+  if (spawn_cmdline ("ostree --repo=repo ls bar /etc/file", &error))
+    g_assert_not_reached ();
+  g_assert (error != NULL);
+  g_clear_error (&error);
+
+  if (spawn_cmdline ("ostree --repo=repo ls bar devnull", &error))
+    g_assert_not_reached ();
+  g_assert (error != NULL);
+  g_clear_error (&error);
+
+ out:
+  g_assert_no_error (error);
+}
+
+static void
+test_libarchive_xattr_callback (gconstpointer data)
+{
+  TestData *td = (void*)data;
+  GError *error = NULL;
+  struct archive *a = archive_read_new ();
+  OstreeRepoImportArchiveOptions opts = { 0 };
+  OstreeRepoCommitModifier *modifier = NULL;
+  char buf[7] = { 0 };
+
+  GVariant* cb (OstreeRepo  *repo,
+                const char  *path,
+                GFileInfo   *file_info,
+                gpointer     user_data)
+  {
```

Comment:
clang doesn't support GNU C nested functions, and I'm kind of with them there.  Instead it implements [blocks](https://en.wikipedia.org/wiki/Blocks_%28C_language_extension%29).